### PR TITLE
Feat: Limiting the number of transactions sent in one request

### PR DESCRIPTION
**Pull request by kristjank**
## Proposed changes
Limiting the size of transaction allowed in one request.

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist
- [x] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
there could be a better place for this thru the API, @faustbrian your feedback needed

### CODE REVIEW

**Comment by faustbrian** [comment]
This should be done in the transaction pool configuration rather then the network configuration.

File: packages/core-api/lib/versions/2/handlers/transactions.js [hunk 1]
```diff
@@ -40,12 +40,17 @@ exports.store = {
...
   async handler (request, h) {
-    if (!transactionPool) {
+    const blockchain = container.resolvePlugin('blockchain')
+    if (!transactionPool || !blockchain) {
       return {
         data: []
       }
     }
 
+    if (request.payload.transactions.length > transactionPool.options.maxTransactionsPerRequest) {
```
Comment (faustbrian) [inline packages/core-api/lib/versions/2/handlers/transactions.js]: This is a v2 endpoint, we are not using those ghetto v1 solutions here.

You will have to update the joi schema to be limited to the configured value.

https://github.com/ArkEcosystem/core/blob/develop/packages/core-api/lib/versions/2/schema/transactions.js#L18

File: packages/core-transaction-pool-redis/lib/defaults.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ module.exports = {
   key: 'ark',
   maxTransactionsPerSender: process.env.ARK_TRANSACTION_POOL_MAX_PER_SENDER || 100,
   allowedSenders: [],
```
Comment (faustbrian) [inline packages/core-transaction-pool-redis/lib/defaults.js]: Missing the `whitelist` option.

**Comment by kristjank** [comment]
@faustbrian

File: packages/core-api/lib/versions/2/handlers/transactions.js [hunk 3]
```diff
@@ -40,17 +40,12 @@ exports.store = {
    * @return {Hapi.Response}
    */
   async handler (request, h) {
-    const blockchain = container.resolvePlugin('blockchain')
-    if (!transactionPool || !blockchain) {
+    if (!transactionPool) {
       return {
```
Comment (faustbrian) [inline packages/core-api/lib/versions/2/handlers/transactions.js]: This should return a `Boom.serverUnavailable('Transaction Pool not ready')`.

File: packages/core-api/lib/versions/2/handlers/transactions.js [hunk 4]
```diff
@@ -40,17 +40,12 @@ exports.store = {
    * @return {Hapi.Response}
    */
   async handler (request, h) {
-    const blockchain = container.resolvePlugin('blockchain')
-    if (!transactionPool || !blockchain) {
+    if (!transactionPool) {
```
Comment (faustbrian) [inline packages/core-api/lib/versions/2/handlers/transactions.js]: Actually this whole block can be removed, the API will only be mounted after the pool and blockchain are ready.

File: packages/core-api/lib/versions/2/handlers/transactions.js [hunk 5]
```diff
@@ -41,7 +41,7 @@ exports.store = {
    */
   async handler (request, h) {
     if (!transactionPool.options.enabled) {
-      return Boom.teapot()
+      return Boom.teapot('Transaction pool is disabled.')
```
Comment (faustbrian) [inline packages/core-api/lib/versions/2/handlers/transactions.js]: I said `serverUnavailable` in my previous comment, not `teapot` as that doesn't give a clear indication of what is wrong just based on the status code.

**Comment by kristjank** [comment]
done as suggested

File: packages/core-api/lib/versions/2/handlers/transactions.js (hunk 5 above)
Comment (kristjank) [reply packages/core-api/lib/versions/2/handlers/transactions.js]: done

File: packages/core-api/lib/versions/2/schema/transactions.js [hunk 6]
```diff
@@ -3,6 +3,9 @@
 const Joi = require('joi')
 const pagination = require('./pagination')
 
+const container = require('@arkecosystem/core-container')
+const transactionPool = container.resolvePlugin('transactionPool')
```
Comment (faustbrian) [inline packages/core-api/lib/versions/2/schema/transactions.js]: If you don't need the full plugin but only it's options you can use `container.resolveOptions('transactionPool')`.

File: packages/core-p2p/lib/server/versions/1/handlers.js [hunk 7]
```diff
@@ -297,13 +297,23 @@ exports.postTransactions = {
    * @return {Hapi.Response}
    */
   async handler (request, h) {
+    const blockchain = container.resolvePlugin('blockchain')
+
+    if (!blockchain) {
+      return h.response({ success: false, error: 'Blockchain not ready' }).code(500)
```
Comment (faustbrian) [inline packages/core-p2p/lib/server/versions/1/handlers.js]: This is already handled by a plugin. https://github.com/ArkEcosystem/core/blob/develop/packages/core-p2p/lib/server/plugins/blockchain-ready.js

File: packages/core-p2p/lib/server/versions/1/handlers.js [hunk 8]
```diff
@@ -297,13 +297,23 @@ exports.postTransactions = {
...
+
     if (!request.payload || !request.payload.transactions || !transactionPool) {
       return {
         success: false,
         transactionIds: []
       }
     }
 
+    if (request.payload.transactions.length > transactionPool.options.maxTransactionsPerRequest) {
+      return h.response({ success: false, error: 'Number of transactions is exceeding max payload size per single request.' }).code(500)
```
Comment (faustbrian) [inline packages/core-p2p/lib/server/versions/1/handlers.js]: This should be broken down into multiple lines to increase readability.

```js
return h.response({
    success: false,
    error: 'Number of transactions is exceeding max payload size per single request.'
}).code(500)
```

File: packages/core/lib/config/testnet/plugins.js [hunk 9]
```diff
@@ -25,7 +25,7 @@ module.exports = {
   },
   '@arkecosystem/core-database-sequelize': {
     dialect: 'sqlite',
-    storage: process.env.ARK_DB_STORAGE || `${process.env.ARK_PATH_DATA}/database/${process.env.ARK_NETWORK_NAME}_.sqlite`,
+    storage: process.env.ARK_DB_STORAGE || `${process.env.ARK_PATH_DATA}/database/${process.env.ARK_NETWORK_NAME}.sqlite`,
```
Comment (faustbrian) [inline packages/core/lib/config/testnet/plugins.js]: This should be reverted as it isn't a config change directly related to this PR.

**Pull request merged**